mockfactory/blinding/catalog.py

```diff
@@ -531,34 +547,62 @@ def Tk(k):
             pphi_prim = 9 / 25 * 2 * np.pi**2 / k**3 * pk_prim(k) / self.cosmo_fid.h**3
             return (pk_lin(k) / pphi_prim)**0.5
 
+        def W(k, sigma=smoothing_radius):
+            return np.exp(- 0.5 * k**2 * sigma**2)
+
         for kslab, slab in zip(mesh.slabs.x, mesh.slabs):
             k = sum(kk.real**2 for kk in kslab)**0.5
             nonzero = k != 0.
-            # minus because applied to randoms
             slab[nonzero] *= bfnl / Tk(k[nonzero])
             slab[~nonzero] = 0.
+
+        if shotnoise_correction:
+
+            # compute the corrective factor at k_pivot
+            mu_pivot = 0.6
+            k_pivot = 4e-3 if _get_from_cosmo(self.cosmo_blind, 'fnl') >= 0 else 8e-3
```

Comment:
maybe it'd better to have this condition on bfnl instead

Reply:
ok